charts/operators-installer/templates/Job_installplan-approver.yaml

```diff
@@ -21,81 +22,49 @@ spec:
-              # NOTE 2: this is nested in {{` `}} so that helm doesn't try to interpret the go template
-              {{`installPlanGoTemplate=$(cat << EOF
-              {{- \$installPlanName := "" -}}
-              {{- range .items -}}
-                {{- \$installPlanItem := . -}}
-                {{- range .spec.clusterServiceVersionNames -}}
-                  {{- if and (eq . "${SUBSCRIPTION_CSV}") (not \$installPlanName) -}}
-                    {{- range \$installPlanItem.metadata.ownerReferences -}}
-                      {{- if eq .uid "${subscriptionUID}" -}}
-                        {{- \$installPlanName = \$installPlanItem.metadata.name -}}
-                      {{- end -}}
-                    {{- end -}}
-                  {{- end -}}
-                {{- end -}}
-              {{- end -}}
-              {{ \$installPlanName }}
-            EOF
-              )`}}
-
-              echo
-              echo "Get InstallPlan for CSV (${SUBSCRIPTION_CSV}) with Subscription (${SUBSCRIPTION_CSV}) (${subscriptionUID}) owner"
-              installPlan=$(oc get installplan -o=go-template="${installPlanGoTemplate}")
-              echo "InstallPlan for CSV (${SUBSCRIPTION_CSV}) with Subscription (${SUBSCRIPTION_CSV}) (${subscriptionUID}) owner: ${installPlan}"
-              if [ ! -z "${installPlan}" ]; then
-                echo
-                echo "Check InstallPlan (${installPlan}) approval"
-                if installPlanApproved=$(oc get installplan ${installPlan} -o=jsonpath="{.spec.approved}"); then
-                  if [ "${installPlanApproved}" == "false" ]; then
-                    echo "Approving InstallPlan (${installPlan})"
-                    oc patch installplan ${installPlan} --type=json -p='[{"op":"replace","path": "/spec/approved", "value": true}]'
-                  else
-                    echo "InstallPlan (${installPlan})already approved"
-                  fi
-                  exit 0
-                else
-                  echo "Failed to look up InstallPlan (${installPlan}) approval"
-                  exit 1
-                fi
-              else
-                echo
-                echo "Could not find InstallPlan for CSV (${SUBSCRIPTION_CSV}) with Subscription (${SUBSCRIPTION_CSV}) (${subscriptionUID}) owner. This can happen if InstallPlan isn't created yet. Try again."
-                exit 1
-              fi
-            else
-              echo
-              echo "Failed to get Subscription ($SUBSCRIPTION) UID. This really shouldn't happen."
-              exit 1
-            fi
+        command: ["/bin/sh","-c"]
+        args:
+          - >-
+              {{- if $.Values.installRequiredPythonLibraries }}
+              python3 -m venv /tmp/venv &&
+              source /tmp/venv/bin/activate &&
+              python3 -m pip install openshift-client semver==2.13.0 --index-url {{ $.Values.pythonIndexURL }} --extra-index-url {{ $.Values.pythonExtraIndexURL }} &&
```

Comment:
@itewk coud we make this whole PR airgap friendly?
I would expect  this chart to be quite useful (At least until OLM 1.0 is GA) on customer envs that are disconnected from internet as well.

Reply:
@jtudelag that is why I put the lib install behind a flag and also gave option to chnage the python lib index. I havnt updated readme yet but I did update the values files with the options for dealing with disconnected envs.

Ideally red hat would have a supported image with the libs in it, but we dont. And it's not a simple matter of making a community image layered on the oc tools image since it's not a ubi and therfor behind the Auth wall.

Hence what I doced in the values file (and will put in the readme). You can either make your own custom image with the right tools/libs, or pull the python packages to a local index (eg nexus or artifactory) and pass tha param to override the index.

Take a look at the values files doc and see if that all makes sense.

Along with putting the doc in the parameters list in readme I intend to make a "disconnected use" section in the readme.

Those updates plus new tests are what I have left to do.

Reply:
Alright, thats more than enough @itewk, I hadn't look deeply into the whole PR, sorry about that.

Reply:
All good. 10 years in NAPS consulting, I wouldn't forget my connection disadvantaged folks :)